documentation/decisions/adr/0007-front-end-language.md

```diff
@@ -0,0 +1,104 @@
+# Front-end Language
+
+- **Status:** Proposed <!-- REQUIRED -->
+- **Last Modified:** 2023-07-10 <!-- REQUIRED -->
+- **Related Issue:** [#96](https://github.com/HHS/grants-api/issues/96) <!-- RECOMMENDED -->
+- **Deciders:** Lucas Brown, Aaron Couch, Billy Daly, Sammy Steiner <!-- REQUIRED -->
+
+## Context and Problem Statement
+
+The goal of this ADR is to select a language that we'll use to implement the front end for beta.grants.gov. The front end will only support static content at first, but will grow to include the new search, supported by the API and eventually the entire grants.gov functionality. Therefore, while a simple solution might work in the short term, it will quickly become insufficient to meet our needs.
+
+## Decision Drivers <!-- RECOMMENDED -->
+
+- Active Maintenance: The language is actively maintained with patches and minor releases delivered on a regular basis
+- Community of Users: The language has an active community of open source users and is commonly used for front-end development
+- Usability: The language is relatively easy to learn for developers without prior experience and there are plenty of resources and training materials available
+- Language Features: The language supports important features like concurrency, static type checking, immutable data types, etc. or there are well established libraries which provide these features
+- Reusability: The language selected for the front end can be also used to support other parts of the project (e.g. ETL & data analysis, API development)
+- HHS and Nava familiarity with the language
+
+## Options Considered
+
+- JavaScript
+- Typescript
+- Python
+- Java
+- Go
+
+## Decision Outcome <!-- REQUIRED -->
+
+Chosen option: Typescript, because it is the ideal language for front-end applications because some form of JavaScript is required for client side functionality anyway. This allows us to reduce the amount of context switching between langauges. Additionally, with node and npm TypeScript and JavaScript are fully featured platforms for building and running applications. We chose TypeScript over JavaScript because of the value that strong typing provides with promomting code quality and developer support. 
```

Comment:
I'm a big TypeScript fan! 

(also, minor nitpick: I think it's stylized with a capital S, just like JavaScript.)

Reply:
BTW, I've also been a big fan of enforcing immutability in JS to reduce bugs, e.g. by using https://immutable-js.com/ and a linter to yell at you if you try to mutate state. But that's a conversation for a later milestone, not this one. Just planting the seed for consideration.

Reply:
Thanks! 
Yes, thanks for catching that. I missed a couple `javascript's too, but it should be good now.